src/mlpack/core/util/param.hpp

```diff
@@ -56,23 +53,117 @@ using DatasetInfo = DatasetMapper<IncrementPolicy, std::string>;
  * PARAM_DOUBLE_OUT_REQ(), PARAM_VECTOR_OUT_REQ(), PARAM_STRING_OUT_REQ().
  *
  * @param NAME Short string representing the name of the program.
+ */
+#define BINDING_PNAME(NAME) static \
+    mlpack::util::ProgramName \
+    io_programname_dummy_object = mlpack::util::ProgramName(NAME);
+
+/**
+ * Document an executable.  Only one instance of this macro should be
+ * present in your program!  Therefore, use it in the main.cpp
+ * (or corresponding executable) in your program.
```

Comment:
(This applies to each of these macros)  Would you mind updating the documentation for this?  e.g., change `Document an executable` to `Specify the short description of a binding`.  (Also we should probably use the word `binding` not `executable` since it's not just command-line programs anymore.) :+1:

Reply:
I have updated the documentation as suggested.